File: tools/improve/tests/devstral-small-2/maudec/py/minimum.py
#
#	<comment from="devstral-small-2">
#	The original code uses a match-case statement to compare two integers N and M and return the smaller one. However, the implementation is redundant because it checks the same condition twice in different cases. We can simplify this by directly comparing N and M and returning the smaller one without using match-case.
#	</comment>
#

def minimum(N: int, M: int) -> int:
    return M if M < N else N

File: tools/improve/tests/devstral-small-2/maudec/cpp/minimum.cpp
//
//	<comment from="devstral-small-2">
//	The original code has redundant conditions: if M <= N, then N <= M is also true, and vice versa. We can simplify by just returning the smaller of the two values.
//	</comment>
//

int minimum(int N, int M) {
  return (M <= N) ? M : N;
}